lib/improver/tests/blending/blend_across_adjacent_points/test_TriangularWeightedBlendAcrossAdjacentPoints.py

```diff
@@ -261,6 +263,57 @@ def test_alternative_parameter_units(self):
         self.assertEqual(self.central_cube.coord('time'), result.coord('time'))
         self.assertArrayAlmostEqual(expected_data, result.data)
 
+    @ManageWarnings(
+        ignored_messages=["Collapsing a non-contiguous coordinate."])
+    def test_works_one_thresh(self):
+        """Test that the plugin retains the single threshold from the input
+           cube."""
+        width = 2.0
+
+        changes = {'points': [0.5], 'units': '1'}
+        cube_with_thresh = add_coord(self.cube.copy(), 'threshold', changes)
+
+        plugin = TriangularWeightedBlendAcrossAdjacentPoints(
+            'forecast_period', self.forecast_period, 'hours', width,
+            'weighted_mean')
+        result = plugin.process(cube_with_thresh)
+
+        # Test that the result cube retains threshold co-ordinates
+        # from origonal cube.
+        self.assertEqual(cube_with_thresh.coord('threshold'),
+                         result.coord('threshold'))
```

Comment:
As there is no unit test to prove that cube_with_thresh has not been changed by the function, we cannot be confident that this assertion is doing what we want.
The easiest solution is to take a copy of cube_with_thresh before running the plugin.process method and do the comparison against the copy.